idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtClassMember.kt

```diff
@@ -0,0 +1,264 @@
+            }
+            else -> null
+        }
+        if (kDoc != null) {
+            newMember.addAfter(kDoc, null)
+        }
+    }
+
+    return newMember
+}
+
+private fun KtAnalysisSession.generateConstructorParameter(
+    project: Project,
+    symbol: KtCallableSymbol,
+    renderOptions: KtDeclarationRendererOptions,
+    isOverride: Boolean,
+): KtCallableDeclaration {
+    return KtPsiFactory(project).createParameter(symbol.render(renderOptions.copy(forceOverrideModifier = isOverride)))
+}
+
+private fun KtAnalysisSession.generateFunction(
+    project: Project,
+    symbol: KtFunctionSymbol,
+    renderOptions: KtDeclarationRendererOptions,
+    bodyType: BodyType,
+    isOverride: Boolean,
+): KtCallableDeclaration {
+    val returnType = symbol.annotatedType.type
+    val returnsNotUnit = !returnType.isUnit
+
+    val body = if (bodyType != BodyType.NO_BODY) {
+        val delegation = generateUnsupportedOrSuperCall(project, symbol, bodyType, !returnsNotUnit)
+        val returnPrefix = if (returnsNotUnit && bodyType.requiresReturn) "return " else ""
+        "{$returnPrefix$delegation\n}"
+    } else ""
+
+    val factory = KtPsiFactory(project)
+    val functionText = symbol.render(renderOptions.copy(forceOverrideModifier = isOverride)) + body
+    return when (symbol) {
+        is KtConstructorSymbol -> {
+            if (symbol.isPrimary) {
+                factory.createPrimaryConstructor(functionText)
+            } else {
+                factory.createSecondaryConstructor(functionText)
+            }
+        }
+        else -> factory.createFunction(functionText)
+    }
+}
+
+private fun KtAnalysisSession.generateProperty(
+    project: Project,
+    symbol: KtPropertySymbol,
+    renderOptions: KtDeclarationRendererOptions,
+    bodyType: BodyType,
+    isOverride: Boolean,
+): KtCallableDeclaration {
+    val returnType = symbol.annotatedType.type
+    val returnsNotUnit = !returnType.isUnit
+
+    val body =
+        if (bodyType != BodyType.NO_BODY) {
+            buildString {
+                append("\nget()")
+                append(" = ")
+                append(generateUnsupportedOrSuperCall(project, symbol, bodyType, !returnsNotUnit))
+                if (!symbol.isVal) {
+                    append("\nset(value) {}")
+                }
+            }
+        } else ""
+    return KtPsiFactory(project).createProperty(symbol.render(renderOptions.copy(forceOverrideModifier = isOverride)) + body)
+}
+
+private fun <T> KtAnalysisSession.generateUnsupportedOrSuperCall(
+    project: Project,
+    symbol: T,
+    bodyType: BodyType,
+    canBeEmpty: Boolean = true
+): String where T : KtNamedSymbol, T : KtTypedSymbol, T : KtCallableSymbol {
+    when (bodyType.effectiveBodyType(canBeEmpty)) {
+        BodyType.EMPTY_OR_TEMPLATE -> return ""
+        BodyType.FROM_TEMPLATE -> {
+            val templateKind = if (symbol is KtFunctionSymbol) TemplateKind.FUNCTION else TemplateKind.PROPERTY_INITIALIZER
```

Comment:
What if it is a KtPropertySymbol or KtEnumEntrySymbol?

Reply:
It's an error in that case. Let me refactor this code to make this intention clearer.